### PR TITLE
added verification and editted the main signing functions a tad

### PR DESCRIPTION
**Pull request by VoR0220**
Signed-off-by: RJ Catalano <richard.catalano@itpeoplecorp.com>

### CODE REVIEW

File: common/libs/signing.ts [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
-import { ecsign, sha3 } from 'ethereumjs-util';
+import { ecsign, ecrecover, publicToAddress } from 'ethereumjs-util';
+import sha3 from 'solidity-sha3';
 import { RawTransaction } from 'libs/transaction';
 import { isValidRawTx } from 'libs/validators';
 
+export interface SignatureArgs {
+  r: Buffer;
+  s: Buffer;
+  v: number;
```
Comment (HenryNguyen5) [inline common/libs/signing.ts]: This should also be a buffer

File: common/libs/signing.ts [hunk 2]
```diff
@@ -26,16 +33,23 @@ export function signMessageWithPrivKey(
   const fullMessage = msg + spacer + date;
   const hash = sha3(fullMessage);
   const signed = ecsign(hash, privKey);
-  const combined = Buffer.concat([
```
Comment (HenryNguyen5) [inline common/libs/signing.ts]: Using https://github.com/MetaMask/eth-simple-keyring/blob/master/index.js#L61 as a reference, they also concentenate `r,s,v` and return the hex string and `0x` prefixed result. Why did this need to be changed?

File: common/libs/signing.ts (hunk 1 above)
Comment (VoR0220) [reply common/libs/signing.ts]: Should it? It appears that in ethereum-js it is returned as a number? Or am i wrong?

File: common/libs/signing.ts (hunk 2 above)
Comment (VoR0220) [reply common/libs/signing.ts]: Easier to handle as an object then having to reparse it from a hex string. Seemed redundant to do so.

**Comment by dternyak** [comment]
We ended up implementing local / offline signing independently as part of our offline-send functionality.

Sorry we weren't able to get this in.

**Comment by VoR0220** [comment]
Thats fair enough. Keep doing what you're doing. I use your common lib from alpha in a ton of my projects...y'all should really turn that into a development boilerplate of some kind...its really useful.

**Pull request closed without merging**